add tests for full screen modal presentation

yanshiyason opened the pull request:
@joemasilotti I took a stab at adding tests for the modal presentation style. Please let me know if this is OK.

yanshiyason committed:
add tests for full screen modal presentation

joemasilotti commented:
Looks good - thank you! Can you please indent the test to match the rest of the file (4 spaces) and add a second "default" test for `.automatic`?

yanshiyason commented:
@joemasilotti Thank you for your review. I fixed the indentation and added a 2nd case for the "default" case.

Something funny happened though.. even though we set it to be `.automatic`, it seems like the value that get's set is actually `.pageSheet`

if you look at the raw enum values, they look like this:

```
-2: automatic
-1: none
0: fullScreen
1: pageSheet
2: formSheet
3: currentContext
```

It seems like (`none` and `automatic`) have negative values, which makes me think that if `.automatic` is set, cocoa switches it for another value depending on some internal logic..

I tested this by momentarily mapping the "default" to `.formSheet`, and the test does indeed pass when I replace `.automatic` with `.formSheet`.

joemasilotti commented:
Nice, thank you!

And I think your understanding is correct. `automatic` is only set and never read. From the docs, it seems like the point of `automatic` is to pick the best style based on the device and orientation.

joemasilotti commented on Tests/TurboNavigatorTests/TurboNavigatorTests.swift:
```diff
@@ -89,6 +89,22 @@ final class TurboNavigatorTests: XCTestCase {
+        let proposal = VisitProposal(context: .modal, modal: .default)
+        navigator.route(proposal)
+
+        XCTAssertEqual(modalNavigationController.modalPresentationStyle, .pageSheet)
```
Let's add a comment as to why we are testing for `.pageSheet` and that behavior might be different for non-iPhone devices.
